### PR TITLE
Provide help on command-line #308

### PR DESCRIPTION
**Pull request by apupier**
it can be useful for external contributor trying to integrate teh Camel
Language Server in another IDE

Signed-off-by: Aurélien Pupier <apupier@redhat.com>

# Pull Request informations

## Rebase & Merge default requirements

1. Green build for master branch
2. Wait 24 hours after PR creation
3. Green job for PR
4. Approved PR

## PR labels default process

- READY_FOR_REVIEW  &rightarrow;  REVIEW_DONE  &rightarrow;  READY_FOR_MERGE

## Tests

- [ ] Are there **Unit tests**?
- [ ] Are there **Integration tests**?
- [ ] Do we need a new **UI test**?

## PR workflow progress

1. [ ] Tagged with relevant **PR labels**
2. [ ] Green **job for PR**
3. [ ] PR was created more than **24 hours ago** or **All committers approved** it
4. [ ] Green **master** branch build

### CODE REVIEW

File: src/main/java/com/github/cameltooling/lsp/internal/Runner.java [hunk 1]
```diff
@@ -39,10 +39,22 @@
 	private static final String PORT_PARAMETER = "--port=";
 	private static final String HOSTNAME_PARAMETER = "--hostname=";
 	private static final String CONTEXTPATH_PARAMETER = "--contextPath=";
+	static final String HELP_PARAMETER = "--help";
+	
+	static final String HELP_MESSAGE =
+			"`--help` to display this message\n\n"
+			+ "When launching without parameter, the Language Client can use Standard Input and Ouput to connect to Camel Language Server.\n\n"
```
Comment (lhein) [inline src/main/java/com/github/cameltooling/lsp/internal/Runner.java]: to the Camel Language Server

File: src/main/java/com/github/cameltooling/lsp/internal/Runner.java [hunk 2]
```diff
@@ -39,10 +39,22 @@
 	private static final String PORT_PARAMETER = "--port=";
 	private static final String HOSTNAME_PARAMETER = "--hostname=";
 	private static final String CONTEXTPATH_PARAMETER = "--contextPath=";
+	static final String HELP_PARAMETER = "--help";
+	
+	static final String HELP_MESSAGE =
+			"`--help` to display this message\n\n"
+			+ "When launching without parameter, the Language Client can use Standard Input and Ouput to connect to the Camel Language Server.\n\n"
+			+ "To use a websocket connection, the parameter `--websocket` must be provided.\n"
+			+ "In this case, 3 other parameters are available:\n"
```
Comment (lhein) [inline src/main/java/com/github/cameltooling/lsp/internal/Runner.java]: I would replace it with "If you are using the websocket connection, 3 additional parameters are available:"

File: src/test/java/com/github/cameltooling/lsp/internal/RunnerHelpTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RunnerHelpTest {
+	
+	private PrintStream sysOut;
```
Comment (lhein) [inline src/test/java/com/github/cameltooling/lsp/internal/RunnerHelpTest.java]: there seems to be an indentation issue

File: src/test/java/com/github/cameltooling/lsp/internal/RunnerHelpTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    
+    @BeforeEach
+    public void beforeClass() {
+    	sysOut = System.out;
```
Comment (lhein) [inline src/test/java/com/github/cameltooling/lsp/internal/RunnerHelpTest.java]: another indentation issue

File: src/test/java/com/github/cameltooling/lsp/internal/RunnerHelpTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        System.setOut(new PrintStream(outContent));
+    }
+    
+    @AfterEach
+    public void afterClass() {
+    	System.setOut(sysOut);
```
Comment (lhein) [inline src/test/java/com/github/cameltooling/lsp/internal/RunnerHelpTest.java]: indent?

File: src/test/java/com/github/cameltooling/lsp/internal/RunnerHelpTest.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    
+	@Test
+	void testClientProxyAvailable() throws Exception {
+		startRunnerWithHelpOption();
+		await("Await that output is written").untilAsserted(() -> assertThat(outContent.toString())
+				.contains(Runner.HELP_MESSAGE));
```
Comment (lhein) [inline src/test/java/com/github/cameltooling/lsp/internal/RunnerHelpTest.java]: indentation issue?

File: src/test/java/com/github/cameltooling/lsp/internal/RunnerHelpTest.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+	}
+
+	private void startRunnerWithHelpOption() {
+		new Thread(new Runnable() {
+			
+			@Override
+			public void run() {
+				Runner.main(new String[]{Runner.HELP_PARAMETER});
+			}
+		}).start();
```
Comment (lhein) [inline src/test/java/com/github/cameltooling/lsp/internal/RunnerHelpTest.java]: indentation issue?

File: src/test/java/com/github/cameltooling/lsp/internal/RunnerStandardIOTest.java [hunk 8]
```diff
@@ -19,15 +19,35 @@
...
+import java.io.PrintStream;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 public class RunnerStandardIOTest {
-
+	
+	private PrintStream sysOut;
```
Comment (lhein) [inline src/test/java/com/github/cameltooling/lsp/internal/RunnerStandardIOTest.java]: indentation

File: src/test/java/com/github/cameltooling/lsp/internal/RunnerStandardIOTest.java [hunk 9]
```diff
@@ -19,15 +19,35 @@
...
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    
+    @BeforeEach
+    public void beforeClass() {
+    	sysOut = System.out;
```
Comment (lhein) [inline src/test/java/com/github/cameltooling/lsp/internal/RunnerStandardIOTest.java]: indentation

File: src/test/java/com/github/cameltooling/lsp/internal/RunnerStandardIOTest.java [hunk 10]
```diff
@@ -19,15 +19,35 @@
...
+        System.setOut(new PrintStream(outContent));
+    }
+    
+    @AfterEach
+    public void afterClass() {
+    	System.setOut(sysOut);
```
Comment (lhein) [inline src/test/java/com/github/cameltooling/lsp/internal/RunnerStandardIOTest.java]: indentation

**Comment by lhein** [comment]
approved once indentation issues are fixed.

File: src/main/java/com/github/cameltooling/lsp/internal/Runner.java [hunk 11]
```diff
@@ -39,10 +39,22 @@
 	private static final String PORT_PARAMETER = "--port=";
 	private static final String HOSTNAME_PARAMETER = "--hostname=";
 	private static final String CONTEXTPATH_PARAMETER = "--contextPath=";
+	static final String HELP_PARAMETER = "--help";
+	
+	static final String HELP_MESSAGE =
+			"`--help` to display this message\n\n"
+			+ "When launching without parameter, the Language Client can use Standard Input and Ouput to connect to the Camel Language Server.\n\n"
+			+ "To use a websocket connection, the parameter `--websocket` must be provided.\n"
+			+ "If you are using the websocket connection, 3 other parameters are available:\n"
```
Comment (bfitzpat) [inline src/main/java/com/github/cameltooling/lsp/internal/Runner.java]: "If you are using the websocket" should be "If you are using a websocket"

File: src/test/java/com/github/cameltooling/lsp/internal/RunnerHelpTest.java [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+	@AfterEach
+	public void afterClass() {
+		System.setOut(sysOut);
+	}
+
+	@Test
+	void testClientProxyAvailable() throws Exception {
+		startRunnerWithHelpOption();
+		await("Await that output is written")
```
Comment (bfitzpat) [inline src/test/java/com/github/cameltooling/lsp/internal/RunnerHelpTest.java]: Maybe "wait for output" would be better?

File: src/test/java/com/github/cameltooling/lsp/internal/RunnerStandardIOTest.java [hunk 13]
```diff
@@ -19,23 +19,45 @@
...
+	public void afterClass() {
+		System.setOut(sysOut);
+	}
+
 	@Test
 	void testClientProxyAvailable() throws Exception {
 		startRunnerWithoutOption();
-		await("Await that Server has started and have a remote proxy client").untilAsserted(() -> assertThat(Runner.server).isNotNull());
-		await("Await that Server has started and have a remote proxy client").untilAsserted(() -> assertThat(Runner.server.getClient()).isNotNull());
+		await("Await that Server has started and have a remote proxy client")
```
Comment (bfitzpat) [inline src/test/java/com/github/cameltooling/lsp/internal/RunnerStandardIOTest.java]: Maybe "Wait for Server to start with a remote proxy client"?

**Pull request merged**